### PR TITLE
Add default method to see if a vector is inside a world border.

### PR DESCRIPTION
**Pull request by m0pt0pmatt**
This adds 2 default methods to check if a vector/location are inside a world border.

I wrote this because I would like to test when things like ender pearls, arrows, and eggs are thrown outside of the world border.

I added a test method which has the same code as the default method (The version of mockito we use doesn't allow for testing default methods).

Let me know if there's anything that needs changing or more testing.

### CODE REVIEW

File: src/test/java/org/spongepowered/api/world/WorldBorderTest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+ */
+package org.spongepowered.api.world;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import com.flowpowered.math.vector.Vector3d;
+import org.junit.Test;
+
+public class WorldBorderTest {
```
Comment (Deamon5550) [inline src/test/java/org/spongepowered/api/world/WorldBorderTest.java]: This unit test is completely useless, sure it helps you validate that your initial logic is correct but it does nothing to test the actual functionality of the class to serve any purpose in the future. All this test will do is get out of date.

Comment (mattysweeps) [reply src/test/java/org/spongepowered/api/world/WorldBorderTest.java]: I agree, but do to limitations with mockito, i can't actually test the method with mocks.

Would you prefer no unit test at all?
Or I could create a large anonymous WorldBorder class and instance?

Comment (Deamon5550) [reply src/test/java/org/spongepowered/api/world/WorldBorderTest.java]: Don't have the unit test at all.

Comment (mbroomfield-r7) [reply src/test/java/org/spongepowered/api/world/WorldBorderTest.java]: @Deamon5550 Unit test has been removed.

**Comment by ryantheleach** [comment]
This works with nether scaling right?

**Comment by m0pt0pmatt** [comment]
@ryantheleach I haven't tested that. I'll do that and get back to you

**Comment by kashike** [comment]
@m0pt0pmatt Do you still wish for this to be pulled? There were no commits in this PR.

**Comment by mattysweeps** [comment]
@kashike This is so old I forgot I had it open. I destroyed the commits by force updating. 

No worries though, if I still need this functionality, I'll make another PR.
I never tested nether scaling anyways.

**Pull request closed without merging**